narwhal/primary/src/aggregators.rs

```diff
@@ -56,12 +63,39 @@ impl VotesAggregator {
             .set(self.votes.len() as i64);
 
         if self.weight >= committee.quorum_threshold() {
-            self.weight = 0; // Ensures quorum is only reached once.
-            return Ok(Some(Certificate::new(
-                committee,
-                header.clone(),
-                self.votes.clone(),
-            )?));
+            let cert = Certificate::new_unverified(committee, header.clone(), self.votes.clone())?;
+            let (_, pks) = cert.signed_by(committee);
+
+            let certificate_digest: Digest<{ crypto::DIGEST_LENGTH }> = Digest::from(cert.digest());
+            match AggregateSignature::try_from(&cert.aggregated_signature)
+                .map_err(|_| DagError::InvalidSignature)?
+                .verify_secure(&to_intent_message(certificate_digest), &pks[..])
+            {
+                Err(err) => {
+                    warn!(
+                        "Failed to verify aggregated sig on certificate: {} error: {}",
+                        certificate_digest, err
+                    );
+                    let mut valid_votes = vec![];
+                    for (pk, sig) in self.votes.clone() {
```

Comment:
not sure how inefficient it is in practice, but note that this function clones each vote 3 times, which could potentially be expensive.

Reply:
good point, revised to remove votes in place instead.